doc/dev/adr/0020-large-message-chunking.md

```diff
@@ -0,0 +1,100 @@
+﻿# ADR 20: Large Message Chunking in MQTT Protocol
+
+## Status
+Proposed
+
+## Context
+The MQTT protocol has inherent message size limitations imposed by brokers and network constraints. Azure IoT Operations scenarios often require transmitting payloads that exceed these limits (e.g., firmware updates, large telemetry batches, complex configurations). Without a standardized chunking mechanism, applications must implement their own fragmentation strategies, leading to inconsistent implementations and interoperability issues.
+
+## Decision
+We will implement sdk-level message chunking as part of the Protocol layer to transparently handle messages exceeding the MQTT broker's maximum packet size.
+
+**The chunking mechanism will**:
+- Be enabled/disabled by a configuration setting.
```

Comment:
Is this actually necessary? It seems like it could be automatic. If the, say, the `TelemetryMessage` payload is too large - just chunk. What's the harm in that? Removes an entire class of conditional error, and allows us to assume input correctness.

What is the argument for toggling it on and off?

Reply:
> so the subscriber should be the SDK as well

We can't assert that. If the receiver doesn't support chunking, then it will receive gibberish when the sender chunks (as opposed to a clear error saying "message too large to send"). And if the sender chunks without the user knowing, that error would be very confusing. I'd generally rather the user opt in to chunking.

Reply:
If this config stays, would it be per envoy? Per message seems excessive, and anything larger in scope doesn't make sense, since different receivers might be configured differently.
Second, if this is an option, I would say it should only be an option on the sending side - the receiving side will either not know what to do with the chunk messages (older version of the SDK or receiver that doesn't use the SDK), or it knows what a chunk message means and might as well reassemble it instead of not getting the message at all